src/farmer.rs

```diff
@@ -138,56 +136,41 @@ impl Builder {
             .map(SingleDiskPlot::wait)
             .collect::<FuturesUnordered<_>>();
 
-        let (cmd_sender, mut cmd_receiver) =
-            futures::channel::mpsc::channel(Self::CMD_CHANNEL_SIZE);
+        let (cmd_sender, cmd_receiver) = oneshot::channel();
 
         if let Some(mut node_runner) = dsn_node_runner {
             tokio::spawn(async move {
                 node_runner.run().await;
             });
         }
 
-        tokio::spawn(async move {
-            let maybe_stop_sender = loop {
-                let cmd = match futures::future::select(
-                    single_disk_plots_stream.next(),
-                    cmd_receiver.next(),
-                )
-                .await
-                {
-                    futures::future::Either::Left((Some(result), _)) => {
-                        result?;
-                        continue;
-                    }
-                    futures::future::Either::Left((None, _)) => return anyhow::Ok(()),
-                    futures::future::Either::Right((None, _)) => break None,
-                    futures::future::Either::Right((Some(cmd), _)) => cmd,
-                };
-
-                match cmd {
-                    FarmerCommand::Stop(stop_sender) => {
-                        break Some(stop_sender);
-                    }
-                }
-            };
+        let handle = tokio::spawn(async move {
+            while let Some(result) = single_disk_plots_stream.next().await {
+                result?;
+            }
+            anyhow::Ok(())
+        });
 
-            drop(single_disk_plots_stream);
-            if let Some(stop_sender) = maybe_stop_sender {
+        tokio::spawn(async move {
+            let maybe_stop_sender = cmd_receiver.await;
+            // TODO: remove once there won't be joining on drop in monorepo
```

Comment:
@i1i1 can you please [create an issue for this](https://github.com/subspace/subspace-sdk/issues/new)? And link it back here please :pray:

Reply:
Here it is #26